### PR TITLE
eth/harness: Work with v1.10.21.

### LINKED ISSUE #1788: server/eth: Add a "consensus layer client" to the simnet harness.
Look under `Non-validating node operators and infrastructure providers` at https://ethereum.org/en/upgrades/merge/

I guess we would use "prysm"? https://docs.prylabs.network/docs/prepare-for-merge

It seems this is necessary for the latest geth release `v1.10.21`

We probably need to have a wiki page for geth server setup as well.

### PR DESCRIPTION
**Pull request by JoeGruffins**
depends on #1785, closes #1788

Originally I thought we would need to add a consensus layer to fix the harness, but it appears we do not. We only needed to keep the authrpc port different. So, I guess this is fine for us. I am not totally sure though.

### CODE REVIEW

File: dex/testing/eth/harness.sh [hunk 1]
```diff
@@ -111,7 +119,7 @@ cat > "${NODES_ROOT}/genesis.json" <<EOF
     "1D4F2ee206474B136Af4868B887C7b166693c194": {
         "balance": "11000000000000000000000"
     },
-    "1c8F908d99F62522F8E0AD9504FFcC187e5514ca": {
+    "946dfaB1AD7caCFeF77dE70ea68819a30acD4577": {
```
Comment (JoeGruffins) [inline dex/testing/eth/harness.sh:122]: Fixes funds for the `simnet-setup.sh` script's seed.

File: dex/testing/eth/harness.sh [hunk 2]
```diff
@@ -19,27 +19,31 @@ ALPHA_ADDRESS_JSON='{"address":"18d65fb8d60c1199bb1ad381be47aa692b482605","crypt
...
 GAMMA_NODE_PORT="30306"
+GAMMA_AUTHRPC_PORT="8554"
 
 DELTA_ADDRESS="d12ab7cf72ccf1f3882ec99ddc53cd415635c3be"
 DELTA_ADDRESS_JSON_FILE_NAME="UTC--2021-03-01T02-31-13.365402148Z--d12ab7cf72ccf1f3882ec99ddc53cd415635c3be"
 DELTA_ADDRESS_JSON='{"address":"d12ab7cf72ccf1f3882ec99ddc53cd415635c3be","crypto":{"cipher":"aes-128-ctr","ciphertext":"a0e9a3da5d0c88c922b5d7e817693552fe17dfd4c598e2a8b08ee53a706a8ffc","cipherparams":{"iv":"28b0a443403b7a02001f07a35724f6e6"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"f790f584bf396cacc06f28201aa697825011e84f570759d6108e20c5ee4fffce"},"mac":"529318e5eec2474221912d01e5a534a0b1dbfb19499ffaf942be6375611caa83"},"id":"d8670e33-8094-45b7-9386-d936e6bf4c1b","version":3}'
 DELTA_NODE_KEY="725394672587b34bbf15580c59e5199c75c2c7e998ba8df3cb38cc4347d46e2b"
 DELTA_ENODE="ca414c361d1a38716170923e4900d9dc9203dbaf8fdcaee73e1f861df9fdf20a1453b76fd218c18bc6f3c7e13cbca0b3416af02a53b8e31188faa45aab398d1c"
 DELTA_NODE_PORT="30307"
+DELTA_AUTHRPC_PORT="8555"
```
Comment (chappjc) [inline dex/testing/eth/harness.sh:46]: geth 1.10.21 tries to bind on a new port now and multiple instances would try to bind the same port?

**Review by chappjc — APPROVED** [review]

File: dex/testing/eth/harness.sh (hunk 1 above)
Comment (chappjc) [reply dex/testing/eth/harness.sh:122]: What broke it?  A previous change of ours or geth 1.10.21?

Comment (JoeGruffins) [reply dex/testing/eth/harness.sh:122]: The eth wallet creation got the extra step where the seed is used to make a mnemonic first. So different address from the same seed.

Comment (chappjc) [reply dex/testing/eth/harness.sh:122]: oh right, you or marton just fixed that in the nodeclient harness code too.

File: dex/testing/eth/harness.sh (hunk 2 above)
Comment (JoeGruffins) [reply dex/testing/eth/harness.sh:46]: Yes, this port option did not exist before. And they all tried to bind to the default.
```
$ geth -help
...
    --authrpc.port value           (default: 8551)
          Listening port for authenticated APIs

```

File: dex/testing/eth/README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ sandboxed environment for testing dex swap transactions.
 ## Dependencies
 
 The harness depends on [geth](https://github.com/ethereum/go-ethereum/tree/master/cmd/geth)
-to run.
+to run. geth v1.10.21+ is recommended.
```
Comment (JoeGruffins) [inline dex/testing/eth/README.md:9]: Snuck this addition in here.

**Review by buck54321 — APPROVED** [review]

**Pull request merged**